### PR TITLE
Embrace Errors

### PR DESCRIPTION
**Pull request by eriktrom**

### CODE REVIEW

File: test/port-finder-z-integration-test.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ vows.describe('portfinder').addBatch({
           timeout = true;
           process.kill(child.pid);
           callback(null, "timeout");
-        }, 10000); // 10 seconds
+        }, 20000); // 10 seconds
```
Comment (ro0gr) [inline test/port-finder-z-integration-test.js]: remove "10 seconds" comment?

Comment (eriktrom) [inline test/port-finder-z-integration-test.js]: thanks @ro0gr 👍

**Comment by eriktrom** [comment]
Checklist 

- [ ] handle when to iterate to next host vs when to iterate to next port

File: lib/portfinder.js [hunk 2]
```diff
@@ -438,8 +446,8 @@ exports._defaultHosts = (function() {
   }
 
   var interfaceNames = Object.keys(interfaces),
-      hiddenButImportantHost = '0.0.0.0', // !important - dont remove, hence the naming :)
-      results = [hiddenButImportantHost];
+      hiddenButImportantHost = ['0.0.0.0', '::'], // !important - dont remove, hence the naming :)
+      results = [].concat(hiddenButImportantHost);
```
Comment (ro0gr) [inline lib/portfinder.js]: Do we need this `concat`? Looks like assignment should be sufficient
```js
results = ['0.0.0.0', '::']
```

**Comment by ro0gr** [comment]
I've tried it locally and have the following error
```
events.js:160
      throw er; // Unhandled 'error' event
      ^

Error: connect EADDRNOTAVAIL 0.0.0.0:4200
    at Object.exports._errnoException (util.js:1022:11)
    at exports._exceptionWithHostPort (util.js:1045:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1087:14)
```

I'm on _win10_. 
`os.networkInterfaces()` does not throw and returns the whole interfaces list.
Hope it helps

**Comment by eriktrom** [comment]
@ro0gr - hey appreciate u taking the time to test it out. Regarding the error with 0.0.0.0 - I'm considering taking that out along with '::'.  Did u try that by any chance?

Either way thanks a bunch for ur feedback.

**Comment by eriktrom** [comment]
closing - will re-open if such codez are needed

**Pull request closed without merging**